### ISSUE TITLE
Add Unacceptable Deadline Conduct Information

### ISSUE THREAD

**Issue by burgess01** [burgess01/code-of-conduct#3]
I had the task of writing the unacceptable deadline section. This is what I came up with:

"
- Students should not leave assignments for the last day.
- Students should not ask for extensions without good reason.
- Students should not contact the professor within 24 hours of a deadline for an extension.
"

I thought the last point was the weakest, with the first being the strongest. I feel like it is important to get your work done early and not leave it for the last minute. Likewise, you should know before the last day before something being due if you will need an extension. 

Should I change the wording of anything? I am trying to explain the idea of proactively working on projects and discouraging people from procrastinating, especially when others depend on them.

**Comment by MAbraham24** [comment]
I feel that the first point is pretty strong in warning students about watching out for deadlines. I would rephrase it by saying "Students should be mindful of assignment due dates and work on them accordingly". This allows students to remember to pace themselves based on their own timeline for working.

**Comment by tuduun** [comment]
I feel like all these bullet points look good!

**Comment by burgess01** [comment]
> I feel that the first point is pretty strong in warning students about watching out for deadlines. I would rephrase it by saying "Students should be mindful of assignment due dates and work on them accordingly". This allows students to remember to pace themselves based on their own timeline for working.

This is a good suggestion, I'll change that right now!

**Comment by mariakimheinert** [comment]
It seems that these changes were merged in PR #7 without my review. Although there are some situations where this would be alright, I worry about the current wording of the Unacceptable Conduct for Deadlines. Specifically, I am concerned that what is written does not align with the course syllabus. Note that the [syllabus](https://mariakimheinert.com/software-engineering/course-syllabus/#assignment-submission) reads:

> Unless otherwise announced, all work will be due on the date and at the time specified in their instructions. Unless special arrangements are made with the instructor, no work will be accepted after the published deadline.

There is no mention of a formal extension policy in the syllabus because extensions are only given in serious and extenuating circumstances. However, the Unacceptable Conduct section seems to suggest that students will receive an extension if they have a "good reason" and privately contact the instructor more than 24 hours in advance, which does not align with the syllabus. Could this section be revised so that it reflects the fact that extensions will only be given in serious and extenuating circumstances and that students should strive always to submit work on time? You may also consider mentioning the resubmission policy that is described in the syllabus. Thoughts?

**Comment by solisa986** [comment]
> It seems that these changes were merged in PR #7 without my review. Although there are some situations where this would be alright, I worry about the current wording of the Unacceptable Conduct for Deadlines. Specifically, I am concerned that what is written does not align with the course syllabus. Note that the [syllabus](https://mariakimheinert.com/software-engineering/course-syllabus/#assignment-submission) reads:
> 
> > Unless otherwise announced, all work will be due on the date and at the time specified in their instructions. Unless special arrangements are made with the instructor, no work will be accepted after the published deadline.
> 
> There is no mention of a formal extension policy in the syllabus because extensions are only given in serious and extenuating circumstances. However, the Unacceptable Conduct section seems to suggest that students will receive an extension if they have a "good reason" and privately contact the instructor more than 24 hours in advance, which does not align with the syllabus. Could this section be revised so that it reflects the fact that extensions will only be given in serious and extenuating circumstances and that students should strive always to submit work on time? You may also consider mentioning the resubmission policy that is described in the syllabus. Thoughts?

@MAbraham24 @tuduun @burgess01

**Comment by laurennevill** [comment]
I think maybe just having one bullet point saying something along the lines of "Students should only request an extension in the case of serious and extenuating circumstances and should do so by communicating with the professor as soon as the circumstances arise." and then adding a whole new point that goes into how there will be an opportunity to resubmit or correct an assignment as is detailed in the syllabus. Thank you for this correction Professor, none of us thought about how our contributions might not correctly align with the syllabus.

**Comment by burgess01** [comment]
Professor, I'm sorry about that! I was told during the lab that I just needed either the TL's or your approval, not both. I can go back and change this though! I can change my section from this to:

- Student should not leave assignments for the last day.
- Students should not request an extension without serious and extenuating circumstances.
- Students who have a situation where they need an extension should contact the professor as soon as possible and not wait until the last minute.

**Comment by laurennevill** [comment]
@burgess01 do you think it would be beneficial to add an extra point about the resubmission opportunity mentioned in the syllabus?

**Comment by burgess01** [comment]
@laurennevill I do, I'm just unsure of how best to phrase it, especially when starting with 'Students should not". Should it be in acceptable deadlines? It can be added to mine, im just not sure how. Do you have any suggestions?

**Comment by laurennevill** [comment]
Hmmm now that you mention that I think it would be better suited for the acceptable deadlines portion.

**Comment by burgess01** [comment]
I created a pull request for the new changes. I'd appreciate more feedback on the pull request, thank you!
